src/components/BookingPage.js

```diff
@@ -0,0 +1,161 @@
+  let seatsForPriceCalc = [];
+  const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+  const seats = [
+    {
+      row: "A",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "B",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "C",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "D",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+  ];
+  let location = useLocation();
+  // console.log(location.state.movie.id, location.state.movie.poster_path);
+
+  function handleSeatClick(id) {
+    if (selectedSeats.includes(id)) {
+      let newSelectedSeats = selectedSeats.filter((seat) => {
+        return id !== seat;
+      });
+      setSelectedSeats(newSelectedSeats);
+    } else if (selectedSeats.length < 10) {
+      setSelectedSeats((selectedSeats) => [...selectedSeats, id]);
+    } else alert("you can not select more than 10 seats");
+  }
+
+  function confirmBooking() {
+    localStorage.setItem(location.state.movie.id, selectedSeats);
+    setConfirmedSeats(
+      localStorage.getItem(location.state.movie.id).split(",").join(", ")
+    );
+    setModalVisible(true);
+    setBookingHomeVisible(false);
+  }
+
+  function closeModal() {
+    setModalVisible(false);
+    setBookingHomeVisible(true);
+  }
+
+  useEffect(() => {
+    if (localStorage.getItem(location.state.movie.id) !== null) {
+      setSelectedSeats(
+        localStorage.getItem(location.state.movie.id).split(",")
+      );
+      setConfirmedSeats(
+        localStorage.getItem(location.state.movie.id).split(",").join(",")
+      );
+    }
+  }, []);
+
+  console.log(location.state.movie.id);
+  return (
+    <>
+      <BookingPageStyle
+        style={
+          bookingHomeVisible
+            ? { background: "white" }
+            : { background: "rgba(229, 229, 229, 0.6)", pointerEvents: "none" }
+        }
+      >
+        <ScreenStyle>
+          <img src={screen}></img>
+        </ScreenStyle>
+
+        <div style={{ margin: "auto" }}>
+          <SeatNumberStyle>
+            {data.map((num) => {
+              return (
+                <SeatStyle>
+                  <h1>{num}</h1>
+                </SeatStyle>
+              );
+            })}
+          </SeatNumberStyle>
+        </div>
+
+        <AllSeatsStyle>
+          {seats.map((item) => {
+            return (
+              <RowSeatStyle>
+                <div>
+                  <h1>{item.row}</h1>
+                </div>
+                {item.seat.map((seatNumber) => {
+                  return (
+                    <>
+                      <SittingSeatStyle
+                        onClick={() => {
+                          handleSeatClick(item.row + seatNumber);
+                        }}
+                      >
+                        {confirmedSeats.includes(item.row + seatNumber) ? (
+                          <img key={item.row + seatNumber} src={greySeat}></img>
```

Comment:
img element must have alt prop. And why key is used here? It should have been used at parent level which is immediate after the map method.